examples/pyo3_benchmarks/src/lib.rs

```diff
@@ -0,0 +1,33 @@
+use pyo3::prelude::*;
```

Comment:
I'm not sure how this benchmark is useful.
It may be difficult to observe some speedup when trying to extract only a few args.

Reply:
Actually although the variance is quite high, there's still a reasonably clear signal as to how good a change is using these benchmarks. I wouldn't take these numbers super seriously for any kind of scientific conclusion. However I can see (for example) that there's still some overhead calling pyo3 functions vs CPython functions, and also that returning `Vec` from `extract_keyword_arguments` leads to noticeable slowdown.

Some example numbers I see at the moment:

```
-------------------------------------------------------------------------------------------- benchmark: 6 tests -------------------------------------------------------------------------------------------
Name (time in ns)                Min                    Max                Mean              StdDev              Median                IQR             Outliers  OPS (Mops/s)            Rounds  Iterations
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
test_no_args_py              60.0000 (1.0)       4,658.0000 (2.86)      67.6030 (1.0)       34.3033 (1.08)      66.0000 (1.0)       2.0000 (inf)      1857;3556       14.7923 (1.0)      149254         100
test_no_args                 70.0000 (1.17)      1,627.0000 (1.0)       78.7283 (1.16)      31.8182 (1.0)       74.0000 (1.12)      2.0000 (inf)      3261;8374       12.7019 (0.86)     128206         100
test_args_and_kwargs_py     219.0476 (3.65)      5,566.6667 (3.42)     245.2637 (3.63)      57.6773 (1.81)     242.8571 (3.68)      4.7619 (inf)     1156;10254        4.0772 (0.28)     196079          21
test_mixed_args_py          288.2353 (4.80)      6,329.4118 (3.89)     320.9264 (4.75)      83.6458 (2.63)     317.6471 (4.81)      5.8824 (inf)     1406;18940        3.1160 (0.21)     192308          17
test_args_and_kwargs        299.9999 (5.00)     86,200.0001 (52.98)    412.2963 (6.10)     466.4672 (14.66)    400.0000 (6.06)      0.0000 (1.0)      359;47106        2.4254 (0.16)     163935           1
test_mixed_args             340.0000 (5.67)     21,400.0000 (13.15)    385.6509 (5.70)     207.9585 (6.54)     380.0000 (5.76)     15.0000 (inf)      1187;1616        2.5930 (0.18)     128206          20
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```

At the moment it's clear we're a bit slower than CPython for function calls, but I hope vectorcall support and #1308 will improve things.

Later I would like to add more numbers to these benchmarks, for example #661.